meson.build

```diff
@@ -0,0 +1,1629 @@
+project('mpv',
+        'c',
+        license: ['GPL2+', 'LGPL2.1+'],
+        version: '0.33',
+        default_options: [
+            'buildtype=debugoptimized',
+            'c_std=c11',
+            'warning_level=1',
+        ]
+)
+
+cc = meson.get_compiler('c')
+
+
+# essential dependencies
+ffmpeg = {
+    'name': 'ffmpeg',
+    'deps': [dependency('libavutil', version: '>= 56.12.100'),
+             dependency('libavcodec', version: '>= 58.12.100'),
+             dependency('libavformat', version: '>= 58.9.100'),
+             dependency('libswscale', version: '>= 5.0.101'),
+             dependency('libavfilter', version: '>= 7.14.100'),
+             dependency('libswresample', version: '>= 3.0.100')],
+}
+
+libass = dependency('libass', version: '>= 0.12.2')
+
+pthreads = dependency('threads')
+
+zlib = cc.find_library('z')
```

Comment:
https://mesonbuild.com/Dependencies.html#zlib

meson special-cases `-lz`, you can use `dependency('zlib')` and it will work everywhere, preferring either pkg-config or cmake if available and falling back to detecting the find_library, handling MSVC-style "zlib1" or "zlib" names on Windows, and checking for the presence of zlib.h too.

Also as a dependency() it can be overridden by a subproject wrap; as a find_library it cannot.